File: 0124-binary-tree-maximum-path-sum/0124-binary-tree-maximum-path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
     private int max_sum = Integer.MIN_VALUE;
    public int maxPathSum(TreeNode root) {
      

         MaxPathSumCore(root);
     return max_sum;


    }

  public int MaxPathSumCore(TreeNode root)
 {
     if (root == null) return 0;

     var leftSum = Math.max(MaxPathSumCore(root.left), 0);
     var rightSum = Math.max(MaxPathSumCore(root.right), 0);

     max_sum = Math.max(max_sum, root.val + leftSum + rightSum);

     return root.val + Math.max(leftSum, rightSum);
 }
}

File: 0124-binary-tree-maximum-path-sum/0124-binary-tree-maximum-path-sum.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
     int maxSum=int.MinValue;
    public int MaxPathSum(TreeNode root) {
        DFS(root);
        return maxSum;
    }

    private int DFS(TreeNode node) {
         if (node == null) return 0;

        // Compute max path sum from left and right subtrees (ignore negatives)
        int left = Math.Max(DFS(node.left), 0);
        int right = Math.Max(DFS(node.right), 0);

        // Calculate the maximum path passing through this node
        int currentPathSum = node.val + left + right;

        // Update the global maximum path sum
        maxSum = Math.Max(maxSum, currentPathSum);

        // Return max sum from this node down to one child
        return node.val + Math.Max(left, right);
    }
}